src/couch/src/couch_passwords.erl

```diff
@@ -41,18 +41,24 @@ hash_admin_password(ClearPassword) when is_binary(ClearPassword) ->
     hash_admin_password(Scheme, ClearPassword).
 
 hash_admin_password("simple", ClearPassword) -> % deprecated
-    Salt = couch_uuids:random(),
+    Salt = salt(),
```

Comment:
This reduces our security. For PBKDF the salt should be (see https://www.owasp.org/index.php/Password_Storage_Cheat_Sheet):
- unique for each stored credential
- generated by cryptographically-strong PRNG

Reply:
@iilyak thanks for pointing out those requirements for the salt. However, those requirements me wonder about the security implications of our current practice of keeping `[couch_httpd_auth] secret` [consistent among nodes](http://docs.couchdb.org/en/latest/setup/cluster.html#preparing-couchdb-nodes-to-be-joined-into-a-cluster), since that value is used in a similar way to generate cookie values. 

Nevertheless, it's pretty clear that this approach will not fly. I think the general approach we would need to take is to obtain the hash prior to setting the admin creds, and then set (`admin-username`, `password-hash`), which won't be changed under the covers. This is sort of the approach taken [here](https://github.com/apache/couchdb/blob/master/src/setup/src/setup.erl#L115) although I believe there's a race because that code doesn't explicitly wait until passwords are hashed, but just gets lucky that the hashing happens quickly enough.

Reply:
@jaydoane Please update testing recommendation.